datalad/distribution/subdatasets.py

```diff
@@ -257,10 +303,36 @@ def _get_submodules(dspath, fulfilled, recursive, recursion_limit,
             # we are not looking for this subds, because it doesn't
             # match the target path
             continue
+        sm.update(modinfo.get(sm['path'], {}))
+        if set_property or delete_property:
+            # do modifications now before we read the info out for reporting
+            # use 'submodule "NAME"' section ID style as this seems to be the default
+            submodule_section = 'submodule "{}"'.format(sm['gitmodule_name'])
+            # first deletions
+            for dprop in assure_list(delete_property):
+                parser.remove_option(submodule_section, dprop)
+                # also kick from the info we just read above
+                sm.pop('gitmodule_{}'.format(dprop), None)
+            # and now setting values
+            for sprop in assure_list(set_property):
+                prop, val = sprop
+                if '{' in val:
```

Comment:
Not to say that we might want to have some property which would contain { in its value and might not want to format it here